packages/teleport/src/Roles/RoleList/RoleListCells.jsx

```diff
@@ -23,17 +23,31 @@ export const RoleNameCell = ({ rowIndex, data }) => {
   return <Cell>{displayName}</Cell>;
 };
 
-export const ActionCell = ({ rowIndex, onEdit, onDelete, data }) => {
+export const ActionCell = ({
+  rowIndex,
+  onEdit,
+  onDelete,
+  data,
+  canDelete,
+  canEdit,
+}) => {
   const { id, owner } = data[rowIndex];
   const onDeleteClick = () => onDelete(id);
   const onEditClick = () => onEdit(id);
+
+  if (!canDelete && !canEdit) {
+    return <Cell align="right" />;
+  }
+
   return (
     <Cell align="right">
       <MenuButton>
-        <MenuItem onClick={onEditClick}>Edit...</MenuItem>
-        <MenuItem disabled={owner} onClick={onDeleteClick}>
-          Delete...
-        </MenuItem>
+        {canEdit && <MenuItem onClick={onEditClick}>Edit...</MenuItem>}
+        {canDelete && (
```

Comment:
I think this will break the auto-close behavior (when canEdit is false, it will render `null`  which MenuButton does not handle correctly when trying to handle click events outside of the menu popup).